policyengine_core/parameters/at_instant_like.py

```diff
@@ -19,3 +19,32 @@ def get_at_instant(self, instant: Instant) -> Any:
 
     @abc.abstractmethod
     def _get_at_instant(self, instant): ...
+
+    def get_attr_dict(self) -> dict:
+        attr_dict = {}
+        attr_list = [
```

Comment:
I think the best practice here would be to keep this method as simple as possible, then do a bit more work with it where we actually use it. What I would do here is:
1. Programmatically loop over all attributes and feed into a relevant data structure
2. Return said structure

I wouldn't recurse downward here, either. I think we can do that inside our custom `ParameterNode` method, where we're more broadly defining what we're doing.